File: js/complete/no013.js
#!/usr/bin/env node

/* Work out the first ten digits of the sum of the following
one-hundred 50-digit numbers. (In data) */

var fns = require('../functions.js'),
    operator = require('../operator.js'),
    timer = require('../timer.js');

exports.main = function() {
    function nonEmptyFilter(str) {
        return (str != '');
    };
    var numbers = fns.getData(13).split('\n').filter(nonEmptyFilter),
        total = operator.bigintSum(numbers);
    return String(total).slice(0, 10);
};

if (require.main === module) {
    timer.timer(13, exports.main);
}

File: python/complete/no013.py
#!/usr/bin/env python

# Work out the first ten digits of the sum of the following
# one-hundred 50-digit numbers. (In data)

from python.decorators import euler_timer
from python.functions import get_data


def main(verbose=False):
    number = get_data(13)
    total = sum(int(line) for line in number.split("\n") if line)
    return str(total)[:10]

if __name__ == '__main__':
    print euler_timer(13)(main)(verbose=True)
